src/components/workspace-utils.js

```diff
@@ -504,3 +505,8 @@ export const NoWorkspacesMessage = ({ onClick }) => {
     ])
   ])
 }
+
+//TODO: Workspace should contain it's cloud provider in a better fashion.
+export const getCloudProviderFromWorkspace = workspace => {
+  return workspace.azureContext ? cloudProviderTypes.AZURE : cloudProviderTypes.GCP
```

Comment:
hmmm... I'd suggest making this typed in `workspace-utils.ts`. You can make a predicate with the `in` and `is` operators in a similar manner to the existing `isGoogleWorkspace`

Something like this should suffice
```
export const isAzureWorkspace = (workspace: BaseWorkspace): workspace is AzureWorkspace => 'azureContext' in workspace
```